binder/summary-statistics.ipynb

```diff
@@ -0,0 +1,110 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from keyname import keyname as kn\n",
+    "import matplotlib.pyplot as plt\n",
+    "import pandas as pd\n",
+    "import seaborn as sns\n",
+    "from teeplot import teeplot\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = pd.read_csv(\"https://osf.io/ck47r/download\")\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "evolutionary_variables = [\n",
+    " 'mut_distn',  # sensitivity analysis\n",
+    " 'num_generations',  # sensitivity analysis\n",
+    " 'num_islands',\n",
+    " 'num_niches',\n",
+    " 'p_island_migration',  # sensitivity analysis\n",
+    " 'p_niche_invasion',  # sensitivity analysis\n",
+    " 'population_size',  # doesn't change\n",
+    " 'tournament_size',\n",
+    "]\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = df[\n",
+    "    (df['subsampling-fraction'] == 1.0) &\n",
+    "    (df['trie-postprocess'] == 'naive')\n",
+    "]\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "stats = df.groupby(evolutionary_variables, as_index = False).agg({\n",
```

Comment:
can you create a "regime" and "quality" column and then groupby that instead? That way we can just have those columns in the table and not the full details of each parameter

you may need to `.dropna(subset=["regime", "quality"])`

```
df["regime"] = df.apply(
    lambda row: {
        (1, 1, 2, 3.0517578125e-08): "plain",
        (1, 1, 1, 3.0517578125e-08): "neutral selection",
        (1, 1, 8, 3.0517578125e-08): "strong selection",
        (1, 4, 2, 3.0517578125e-06): "weak 4 niche ecology",
        (1, 4, 2, 3.0517578125e-08): "4 niche ecology",
        (1, 8, 2, 3.0517578125e-08): "8 niche ecology",
        (1024, 1, 2, 3.0517578125e-08): "spatial structure",
    }.get(
        tuple(
            row[
                [
                    "num_islands",
                    "num_niches",
                    "tournament_size",
                    "p_niche_invasion",
                ]
            ]
        ),
        np.nan,  # default
    ),
    axis="columns",
)
```

```
df["quality"] = df.apply(
    lambda row: {
        ("reconstructed-tree", 3.0): "33% resolution",
        ("reconstructed-tree", 10.0): "10% resolution",
        ("reconstructed-tree", 30.0): "3% resolution",
        ("reconstructed-tree", 100.0): "1% resolution",
        ("collapsed-phylogeny", 0): "perfect resolution",
    }.get(
        tuple(row[["a", "resolution"]].fillna(0)),
        np.nan,  # default
    ),
    axis="columns",
)
df = df.dropna(axis="index", subset=["quality"]).copy()
df["quality"].unique()
```

also, for ease of use later, I think we should add a column that summarizes all sensitivity analysis variables (i.e., appends them all into a single keyname string). That will make our lives much easier if we want to filter the table, i.e., to only show the main condition in one table and the other sensitivity analysis conditions in another,

```
sensitivity_analysis_variables = [
    "epoch",
    "mut_distn",
]
```

Reply:
Done in 3702606